### ISSUE TITLE
NullPointerException on startup

### ISSUE THREAD

**Issue by andreas-eberle** [jsettlers/settlers-remake#240]
Sebastian send me this by mail: 

I just wanted to try out your settlers 3 remake and got the following
Exception:

g:\Spiele\JSettlers>java -jar JSettlers.jar
Exception in thread "main" java.lang.NullPointerException
        at sun.awt.shell.Win32ShellFolder2.getFileSystemPath(Unknown
Source) at sun.awt.shell.Win32ShellFolder2.access$400(Unknown Source)
        at sun.awt.shell.Win32ShellFolder2$10.call(Unknown Source)
        at sun.awt.shell.Win32ShellFolder2$10.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown
Source) at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown
Source) at
sun.awt.shell.Win32ShellFolderManager2$ComInvoker$3.run(Unknown Sourc e)
        at java.lang.Thread.run(Unknown Source)

**Comment by andreas-eberle** [comment]
I requested further information about OS type/version, used game release and Java version.

**Comment by andreas-eberle** [comment]
This might be caused by http://bugs.java.com/bugdatabase/view_bug.do?bug_id=6973685

**Comment by UnitedHaxor** [comment]
Hi,
I've used https://github.com/jsettlers/settlers-remake/releases/download/v0.1.3-alpha/JSettlers_v0.1.3-alpha.zip

I'm using Windows 7 64-bit and Java8:

D:\Users\United>java -version
java version "1.8.0_51"
Java(TM) SE Runtime Environment (build 1.8.0_51-b16)
Java HotSpot(TM) 64-Bit Server VM (build 25.51-b03, mixed mode)

But it does not work with Java 7 too:
E:\Java\jre7\bin>java -version
java version "1.7.0_45"
Java(TM) SE Runtime Environment (build 1.7.0_45-b18)
Java HotSpot(TM) 64-Bit Server VM (build 24.45-b08, mixed mode)

E:\Java\jre7\bin>java -jar c:\JSettlers\JSettlers.jar
Exception in thread "main" java.lang.NullPointerException
        at sun.awt.shell.Win32ShellFolder2.getFileSystemPath(Unknown Source)
        ...
        at java.lang.Thread.run(Unknown Source)


I also disabled my anti virus scanner and my firewall, but nothing helped.

**Comment by UnitedHaxor** [comment]
BTW: It works with Xubuntu 15.04 4 bit

**Comment by tomsoftware** [comment]
I've seen this bug before and may this is a internal FileDialog() bug and we could use FileDialog() instead?

@UnitedHaxor : I add a small Test Java Class... may you can test this on you Win7 System?

http://www.file-upload.net/download-11047066/test.zip.html

- the "testok.java" uses FileDialog() and should work
- the "testfail.java" uses the JFileChooser() and may fail

use 
 >  java testfail

and

 >  java testok

for runing / testing

**Comment by andreas-eberle** [comment]
@UnitedHaxor: Can you try the tests of @tomsoftware?
Furthermore, can you try to log in as a different user and run the tests or JSettlers with that user? Does this help? Does it help to update your Java 8 to the newest version? 

This most likely is a Java bug. It seems they fixed it once (http://bugs.java.com/bugdatabase/view_bug.do?bug_id=6544857) but they also have a ticket with the same issue (as far as I can see) which is still open (http://bugs.java.com/bugdatabase/view_bug.do?bug_id=6973685).

**Comment by UnitedHaxor** [comment]
I've just tried both test applications and the result is as you expected. testok works and testfail does not.
I also tried JSettlers.jar running as Administrator, but it shows me the same NPE.
After I updated my java to
>java -version
java version "1.8.0_65"
Java(TM) SE Runtime Environment (build 1.8.0_65-b17)
Java HotSpot(TM) 64-Bit Server VM (build 25.65-b01, mixed mode)

I tried it again and testfail and JSettlers do still not work.
